src/migrations/index.js

```diff
@@ -0,0 +1,48 @@
+import database from '../databases';
+import log from '../log';
+import populateUploadedAt from './populateTokensUploadedAt';
+
+// eslint-disable-next-line no-process-env
+const APP_INSTANCE = process.env.NODE_APP_INSTANCE;
+
+function canNodeRunMigration() {
+  // Run when not in PM2 or on the first instance only
+  return !APP_INSTANCE || APP_INSTANCE === '0';
+}
+
+function startMigration(method) {
+  return new Promise(resolve => {
+    if (!canNodeRunMigration()) {
+      resolve();
+      return;
+    }
+    const reschedule = () => setTimeout(async () => {
+      const done = await method();
+      if (!done) {
+        reschedule();
+      } else {
+        resolve();
+      }
+    }, 2500);
```

Comment:
Imo it's not the responsibility of this logic to assume a migration function is only a partial migration. The migration function should resolve iff it's actually complete, and handle any batching (which this seems to do) internally. I would expect any failure to result in a rejected promise.